control-plane/roles/gardener/defaults/main/extensions.yaml

```diff
@@ -1,6 +1,7 @@
 ---
 gardener_extension_networking_calico_enabled: true
 gardener_extension_networking_cilium_enabled: true
+gardener_extension_networking_cilium_vector_overwrite_image_tag: v1.12.1
```

Comment:
To reduce this maintenance overhead I propose adding the variable `gardener_extension_networking_cilium_image_vector_overwrite`. By default this is empty and no overwrite is being made. This way the extension just deploys the image vector bundled together with the extension. Operators can provide an overwrite completely on their own.

We will declare this a a breaking change.